src/components/FindMovie/FindMovie.tsx

```diff
@@ -1,57 +1,124 @@
-import React from 'react';
+import React, { useCallback, useState } from 'react';
+import cn from 'classnames';
 import './FindMovie.scss';
+import { MovieCard } from '../MovieCard';
+import { getMovie } from '../../api';
+import { Movie } from '../../types/Movie';
+import { MovieData } from '../../types/MovieData';
+import { mapMovie } from '../../utils/mapMovie';
+
+type Props = {
+  onAddMovie: (movie: Movie) => void;
+};
+
+export const FindMovie: React.FC<Props> = ({ onAddMovie }) => {
+  const [query, setQuery] = useState('');
+  const [movie, setMovie] = useState<Movie | null>(null);
+  const [loading, setLoading] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const showError = useCallback((message: string) => {
+    setErrorMessage(message);
+  }, []);
+
+  const handleSubmit = async (e: React.FormEvent) => {
+    e.preventDefault();
+    if (!query.trim()) {
+      return;
+    }
+
+    setLoading(true);
+    setMovie(null);
+    setErrorMessage('');
+
+    try {
+      const data = await getMovie(query);
+
+      if ('Response' in data && data.Response === 'False') {
+        showError("Can't find a movie with such a title");
+      } else {
+        setMovie(mapMovie(data as MovieData));
```

Comment:
You should fix getMovie - the result should be cast to the correct type inside
Or you need to check for the type ResponceError.

Anyways - your final result should not include excplicit casting

```suggestion
        setMovie(mapMovie(data));
```

Reply:
> You should fix getMovie - the result should be cast to the correct type inside Or you need to check for the type ResponceError.
> 
> Anyways - your final result should not include excplicit casting

Thanks for highlight, fixed in [02a5fb8](https://github.com/mate-academy/react_movies-list-fetch-movies/pull/1686/commits/02a5fb8a4c4fd5d0a277dcd85b34f7063ace4861)